designs/timeout-api.md

```diff
@@ -0,0 +1,78 @@
+---
+state: draft
+---
+
+# Node Timout API
+
+## Summary
+
+This design looks at how to add a timeout mechanism to nodes. This will allow
+flows to be created that can handle hangs or other unexpected events that prevent
+the normal operation of a flow.
+
+## Authors
+
+ - @knolleary
+
+## Details
+
+This design builds on the new [Messsaging API design](node-messaging-api) that
+introduces the `node.done` function for a node to indicate it has completed
+handling a message.
+
+It will be possible for the runtime to timeout a node that does not complete its
+processing of a message within a given time. But this can only work if the runtime
+knows a node has been updated to call `node.done`.
+
+**By default, a Node cannot be timed out**
+
+It will be up to individual implementations to decide if it makes sense to add
+timeout handling. Some nodes, such as the Change and Switch nodes do not perform
```

Comment:
Not important at this point, but `Change` and `Switch` nodes can take a long time because of use of context store.